independent-projects/tools/common/src/main/java/io/quarkus/maven/utilities/MojoUtils.java

```diff
@@ -40,39 +37,77 @@
     public static final String KOTLIN_EXTENSION_NAME = "kotlin";
     public static final String SCALA_EXTENSION_NAME = "scala";
 
-    private static final String PLUGIN_VERSION_PROPERTY_NAME = "quarkus.version";
-    public static final String QUARKUS_VERSION_PROPERTY = "${" + PLUGIN_VERSION_PROPERTY_NAME + "}";
+    public static final String QUARKUS_VERSION_PROPERTY_NAME = "quarkus.version";
+    public static final String QUARKUS_VERSION_PROPERTY = "${" + QUARKUS_VERSION_PROPERTY_NAME + "}";
 
-    private static final Properties properties = new Properties();
+    private static Properties properties;
 
-    private MojoUtils() {
-        // Avoid direct instantiation
+    private static QuarkusPlatformDescriptor platformDescr;
+
+    private static QuarkusPlatformDescriptor getPlatformDescriptor() {
+        return platformDescr == null ? platformDescr = QuarkusPlatformConfig.getGlobalDefault().getPlatformDescriptor() : platformDescr;
+    }
+
+    private static Properties getProperties() {
+        if(properties == null) {
+            try {
+                properties = getPlatformDescriptor().loadResource("quarkus.properties", is -> {
+                    final Properties props = new Properties();
+                    props.load(is);
+                    return props;
+                });
+            } catch (IOException e) {
+                throw new IllegalStateException("The quarkus.properties file cannot be read", e);
+            }
+        }
+        return properties;
     }
 
-    static {
-        loadProperties();
+    private MojoUtils() {
+        // Avoid direct instantiation
     }
 
     public static Map<String, String> getAllProperties() {
         Map<String, String> all = new HashMap<>();
-        properties.stringPropertyNames().forEach(s -> all.put(s, properties.getProperty(s)));
+        getProperties().stringPropertyNames().forEach(s -> all.put(s, getProperties().getProperty(s)));
         return all;
     }
 
     public static String getPluginArtifactId() {
-        return get("plugin-artifactId");
+        return "quarkus-maven-plugin";
```

Comment:
is this really right? same question on the next 2 getters....why isn't this resolving the properties rather than being hardcoded ?